tests/test_settings.py

```diff
@@ -1993,3 +2008,557 @@ class Settings(BaseSettings):
+"""
+        )
+
+
+def test_cli_enforce_required(env):
+    class Settings(BaseSettings):
+        my_required_field: str
+
+    env.set('MY_REQUIRED_FIELD', 'hello from environment')
+
+    assert Settings(_cli_parse_args=[], _cli_enforce_required=False).model_dump() == {
+        'my_required_field': 'hello from environment'
+    }
+
+    with pytest.raises(SystemExit):
+        Settings(_cli_parse_args=[], _cli_enforce_required=True).model_dump()
+
+
+@pytest.mark.parametrize(
+    'value,expected',
+    [
+        (str, 'str'),
+        ('foobar', 'str'),
+        ('SomeForwardRefString', 'str'),  # included to document current behavior; could be changed
+        (List['SomeForwardRef'], "List[ForwardRef('SomeForwardRef')]"),  # noqa: F821
+        (Union[str, int], '{str,int}'),
+        (list, 'list'),
+        (List, 'List'),
+        ([1, 2, 3], 'list'),
+        (List[Dict[str, int]], 'List[Dict[str,int]]'),
+        (Tuple[str, int, float], 'Tuple[str,int,float]'),
+        (Tuple[str, ...], 'Tuple[str,...]'),
+        (Union[int, List[str], Tuple[str, int]], '{int,List[str],Tuple[str,int]}'),
+        (foobar, 'foobar'),
+        (LoggedVar, 'LoggedVar'),
+        (LoggedVar(), 'LoggedVar'),
+        (Representation(), 'Representation()'),
+        (typing.Literal[1, 2, 3], '{1,2,3}'),
+        (typing_extensions.Literal[1, 2, 3], '{1,2,3}'),
+        (SimpleSettings, 'JSON'),
+        (Union[SimpleSettings, SettingWithIgnoreEmpty], 'JSON'),
+        (Union[SimpleSettings, str, SettingWithIgnoreEmpty], '{JSON,str}'),
+        (Union[str, SimpleSettings, SettingWithIgnoreEmpty], '{str,JSON}'),
+    ],
+)
+def test_cli_metavar_format(value, expected):
+    assert CliSettingsSource(SimpleSettings)._metavar_format(value) == expected
+
+
+@pytest.mark.skipif(sys.version_info < (3, 10), reason='requires python 3.10 or higher')
+@pytest.mark.parametrize(
+    'value_gen,expected',
+    [
+        (lambda: str, 'str'),
+        (lambda: 'SomeForwardRefString', 'str'),  # included to document current behavior; could be changed
+        (lambda: List['SomeForwardRef'], "List[ForwardRef('SomeForwardRef')]"),  # noqa: F821
+        (lambda: str | int, '{str,int}'),
+        (lambda: list, 'list'),
+        (lambda: List, 'List'),
+        (lambda: list[int], 'list[int]'),
+        (lambda: List[int], 'List[int]'),
+        (lambda: list[dict[str, int]], 'list[dict[str,int]]'),
+        (lambda: list[Union[str, int]], 'list[{str,int}]'),
+        (lambda: list[str | int], 'list[{str,int}]'),
+        (lambda: LoggedVar[int], 'LoggedVar[int]'),
+        (lambda: LoggedVar[Dict[int, str]], 'LoggedVar[Dict[int,str]]'),
+    ],
+)
+def test_cli_metavar_format_310(value_gen, expected):
+    value = value_gen()
+    assert CliSettingsSource(SimpleSettings)._metavar_format(value) == expected
+
+
+@pytest.mark.skipif(sys.version_info < (3, 12), reason='requires python 3.12 or higher')
+def test_cli_metavar_format_type_alias_312():
+    exec(
```

Comment:
Is this OK? I didn't know how else to get coverage on the `TypeAliasType` without breaking earlier versions of Python (i.e. `type TypeAliasInt = int` is only available starting in Python 3.12), so used exec here to avoid an interpreter error.